json/src/test/scala-2.12/com/gu/contentapi/json/CirceRoundTripSpec.scala

```diff
@@ -166,20 +167,29 @@ class CirceRoundTripSpec extends FlatSpec with Matchers {
     checkRoundTrip[EntitiesResponse]("entities.json")
   }
 
-  def checkRoundTrip[T : Decoder : Encoder](jsonFileName: String,
-                                  transformBeforeDecode: Json => Json = identity,
-                                  transformAfterEncode: Json => Json = identity) = {
-
-    val jsons: Option[(Json, Json)] = for {
-      jsonBefore <- parse(loadJson(jsonFileName)).toOption
-      transformedBefore <- jsonBefore.hcursor.downField("response").success.map(c => transformBeforeDecode(c.value))
-      decoded <- transformedBefore.as[T].toOption
+  def checkRoundTrip[T : Decoder : Encoder](
+    jsonFileName: String,
+    transformBeforeDecode: Json => Json = identity,
+    transformAfterEncode: Json => Json = identity
+  ) = {
+    val jsonBefore: Json = parse(loadJson(jsonFileName)).toOption.get
+    val transformedBefore: Json = jsonBefore.hcursor.downField("response").success.map(
+      c => transformBeforeDecode(c.value)
+    ).get
```

Comment:
do we definitely want these two lines to throw an exception? I know it's only a test but at the same time we are going through effort later to provide a nicer failure if the second part of the round trip fails.

Reply:
Fair enough, it was mostly laziness! I’ll update

Reply:
Have added [commit 9c266ba](https://github.com/guardian/content-api-models/pull/239/commits/9c266ba8a0c23a7813f8711e69c503a807ce5f46) to handle these error cases more nicely